zebrad/src/components/mempool.rs

```diff
@@ -134,6 +140,13 @@ impl Service<Request> for Mempool {
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        // Clear the mempool if there has been a chain tip reset.
+        if let Some(Ok(TipAction::Reset { height: _, hash: _ })) =
+            self.chain_tip_change.tip_change().now_or_never()
```

Comment:
This code is just about right, but we don't want to have to wait for the next change here.

So I think `ChainTipChange` needs a sync method that returns an `Option<TipAction>`:
* `Some(_)` if there has been a change since the last time the method was called
* `None` if there has been no change

I've run out of time today, but I can submit a PR tomorrow.

Reply:
I'll try to write it.

Reply:
Won't `now_or_never()` do exactly that? From what I understand it won't wait for the next change.
(I'm interested in this because https://github.com/ZcashFoundation/zebra/issues/2711 will be similar)

Reply:
My guess is that Teor just meant to refactor the code as in https://github.com/ZcashFoundation/zebra/pull/2777.

Reply:
> Won't `now_or_never()` do exactly that?

Nice catch! I had planned a much more complicated refactor.

There's just a few tricky edge cases we need to handle, see #2777 for details.

Reply:
Resolved by #2777.